src/scconf/scconf.c

```diff
@@ -411,6 +411,8 @@ char *scconf_list_strdup(const scconf_list * list, const char *filler)
 	if (filler) {
 		len += scconf_list_array_length(list) * (strlen(filler) + 1);
 	}
+	if (len == 0)
+		return NULL;
 	buf = malloc(len);
```

Comment:
according to the Linux manual, calling `malloc` with `0` should be fine, but that may be different on other platforms.